app/javascript/packages/document-capture/components/form-steps.jsx

```diff
@@ -197,6 +197,12 @@ function FormSteps({
   const { form: Form, footer: Footer, name, title } = step;
   const isLastStep = stepIndex + 1 === steps.length;
 
+
+  // fields.current doesn't contain the current step's field until after rendering, because of registerField()
+  const canContinue = Object.keys(values).length &&
+    Object.keys(fields.current).every(field => !!values[field]) &&
+    !activeErrors.length;
```

Comment:
@aduth I dropped this as a WIP, was hoping to get some feedback from you on how to figure out if a form is "done"

- My goal was to express something like "we have a value for every field and no errors"
- As far as I can tell, we don't know a field exists until after `registerField` is called, so for the first render on the second page, we "have all the fields" because the selfie field hasn't been registered yet, so the button is not disabled as expeted

I am thinking we might want a way to fields a priori, but then have a way to "attach" them to their DOM elements, so maybe rename `registerField` to something like `linkField` or `attachField`?

Reply:
>My goal was to express something like "we have a value for every field and no errors"

The logic in `toNextStep` (specifically `getValidationErrors`, which also accounts for a "`required`" property of a field, of which all of ours are required) could be used as reference, or directly reused in the case of `getValidationErrors`.

> * As far as I can tell, we don't know a field exists until after `registerField` is called, so for the first render on the second page, we "have all the fields" because the selfie field hasn't been registered yet, so the button is not disabled as expeted

Yeah, that's tricky. One thing I wonder though is... in practice, does it actually perceivable as being enabled? Or does it happen quickly enough that it's not a practical concern? 

If we needed to, we could make the behavior less of a "push from the `step` to the `FormSteps`" and more a "pull from the `step` at the `FormSteps`", like a `getFields` static member. I seem to recall some prior work around this or something similar.

> I am thinking we might want a way to fields a priori, but then have a way to "attach" them to their DOM elements, so maybe rename `registerField` to something like `linkField` or `attachField`?

Not sure I fully understand what's being proposed here, but we do assign the underlying DOM element for each field:

https://github.com/18F/identity-idp/blob/bb33b8a6508dcbe215f49b18b386c051fad1fc48/app/javascript/packages/document-capture/components/form-steps.jsx#L229

Or do you mean something like a combination of the above ("basic field metadata") and making this more a separate step of associating that DOM element with the field.

Reply:
I meant having basic field metadata ahead of time, so we don't want to wait on the render